### PR TITLE
PP-3701/Refactor Middlewares

### PR DESCRIPTION
**Pull request by ckalista**
## WHAT

* Renamed former `get-payment-request` middleware to `check-secure-cookie` and narrowed down its purpose to only checking that the payment request externalId has been set in the session for security reasons. If not, the middleware interrupts the flow with an error. 
* Repurposed former `get-payment-request` to retrieve the payment request from connector and expose the data through `res.locals`.
* With `get-payment-request` middleware in place we no longer need to have the payment request in the session. Refactored the `secure` controller not to save the payment request in the session. Refactored all other controllers to retrieve the payment request from the `res.local.paymentRequest` instead of getting it from the session.

### CODE REVIEW

File: app/cancel/get.controller.js [hunk 1]
```diff
@@ -5,12 +5,15 @@ const connectorClient = require('../../common/clients/connector-client')
 const {getSessionVariable} = require('../../common/config/cookies')
 
 module.exports = (req, res) => {
-  const paymentRequestExternalId = req.params.paymentRequestExternalId
-  const paymentRequest = getSessionVariable(req, paymentRequestExternalId).paymentRequest
+  const paymentRequest = res.locals.paymentRequest
+  const paymentRequestExternalId = paymentRequest.externalId
+  const gatewayAccountExternalId = paymentRequest.gatewayAccountExternalId
+  const returnUrl = paymentRequest.returnUrl
```
Comment (jonheslop) [inline app/cancel/get.controller.js]: This looks great but performance wise it’s maybe not worth setting a const for something that is only used once. @brid is this true? Maybe could just set `paymentRequest` and the rest come from that one?

Comment (ckalista) [reply app/cancel/get.controller.js]: Yes, agreed. I have changed that.

Comment (brid) [reply app/cancel/get.controller.js]: performance wise, you can argue that 'const' leaves things up to the garbage collector to dispose of references against which in the case of things holding onto functions/closures or other complexity can degrade performance compared to using a 'let' and explicitly setting this to undefined.

Obviously this depends on how hard you are hitting a certain piece of code, in what circumstances, on what hardware etc etc etc.

File: app/cancel/get.controller.tests.js [hunk 2]
```diff
@@ -36,8 +41,15 @@ describe('cancel GET controller', () => {
 
   describe('when cancelling a payment journey', () => {
     before(done => {
-      nock(config.CONNECTOR_URL).post(`/v1/api/accounts/${paymentRequest.gatewayAccountExternalId}/payment-requests/${paymentRequestExternalId}/cancel`).reply(200)
-      nock(config.CONNECTOR_URL).get(`/v1/api/accounts/${paymentRequest.gatewayAccountExternalId}`).reply(200, gatewayAccount)
+      nock(config.CONNECTOR_URL)
```
Comment (DanailMinchev) [inline app/cancel/get.controller.tests.js]: We need to migrate to Pact at some point

File: package-lock.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "@sinonjs/formatio": {
       "version": "2.0.0",
       "resolved": "http://registry.npmjs.org/@sinonjs/formatio/-/formatio-2.0.0.tgz",
-      "integrity": "sha512-ls6CAMA6/5gG+O/IdsBcblvnd8qcO/l1TYoNeAzp3wcISOxlPXQEus0mLcdwazEkWjaBdaJ3TaxmNgCLWwvWzg==",
+      "integrity": "sha1-hNt+nrVTHfGKjF4L+25EnlXmVLI=",
```
Comment (DanailMinchev) [inline package-lock.json]: We don't have any `package.json` changes, so no need to modify `package-lock.json`

**Pull request merged**